modules/deployment/src/main/java/org/eclipse/xpanse/modules/deployment/ServiceConfigurationManager.java

```diff
@@ -241,4 +250,91 @@ private void validate(ServiceTemplateEntity serviceTemplateEntity,
                 configurationParameters, serviceConfigurationUpdate.getConfiguration(),
                 jsonObjectSchema);
     }
+
+    /**
+     * Query pending configuration change request for agent.
+     */
+    public ResponseEntity<ConfigurationChangeRequestForAgent>
+            getPendingConfigurationChangeRequest(String serviceId, String resourceName) {
+        ServiceConfigurationUpdateRequestQueryModel model =
+                new ServiceConfigurationUpdateRequestQueryModel(null, UUID.fromString(serviceId),
+                        resourceName, null, ServiceConfigurationStatus.PENDING);
+        List<ServiceConfigurationUpdateRequest> requests =
+                serviceConfigurationUpdateStorage.listServiceConfigurationUpdateRequests(model);
+        if (CollectionUtils.isEmpty(requests)) {
+            return ResponseEntity.status(HttpStatus.NO_CONTENT)
+                    .body(new ConfigurationChangeRequestForAgent());
+        }
+        Optional<ServiceConfigurationUpdateRequest> latestRequest = requests.stream()
+                .filter(request -> Objects.nonNull(request.getServiceOrderEntity())
+                        && Objects.nonNull(request.getServiceOrderEntity().getStartedTime()))
+                .min(Comparator.comparing(request ->
```

Comment:
I think the `startedTime` of  latest configuration request in the list should have the maximum value. Please check this.

Reply:
![image](https://github.com/user-attachments/assets/013801ba-7c18-4d53-b1b3-5838fcc6a168)
Hi ,We need to return the oldest data here, not the latest